base/ca/src/main/java/com/netscape/ca/CertificateAuthority.java

```diff
@@ -1564,16 +1562,16 @@ public OCSPResponse validate(OCSPRequest request)
             Request req = tbsReq.getRequestAt(0);
             BigInteger serialNo = req.getCertID().getSerialNumber();
 
-            CertificateRepository certificateRepository = engine.getCertificateRepository();
-            X509CertImpl cert = certificateRepository.getX509Certificate(serialNo);
-
-            X500Name certIssuerDN = (X500Name) cert.getIssuerDN();
-            ocspCA = engine.getCA(certIssuerDN);
-        }
+            try {
+                CertificateRepository certificateRepository = engine.getCertificateRepository();
+                X509CertImpl cert = certificateRepository.getX509Certificate(serialNo);
 
-        if (ocspCA == null) {
-            logger.error("CertificateAuthority: Could not locate issuing CA");
-            throw new CANotFoundException("Could not locate issuing CA");
+                X500Name certIssuerDN = (X500Name) cert.getIssuerDN();
+                ocspCA = engine.getCA(certIssuerDN);
+            } catch (EBaseException e) {
+                // If we don't know the issuer allow this CA to validate
+                // and report the CertStatus as Unknown
+            }
```

Comment:
I think here we should catch `EDBRecordNotFoundException` specifically (see `LDAPSession.read()`) which indicates that the cert doesn't exist in CA's cert repository. Other exceptions might indicate there's a real issue in CA so we should let it bubble up like before so it can be handled properly by the caller.